sopel/plugins/exceptions.py

```diff
@@ -16,3 +16,12 @@ def __init__(self, name):
         message = 'Plugin "%s" not registered' % name
         self.plugin_name = name
         super(PluginNotRegistered, self).__init__(message)
+
+
+class ImproperlyConfigured(PluginError):
```

Comment:
I'm confused as to why we need to add this when `sopel.config.ConfigurationError` [already exists](https://github.com/sopel-irc/sopel/blob/a5a6e2dcf3ad18e582ed063a41670106c6fc2be5/sopel/config/__init__.py#L78)? Since the error is ultimately due to a missing value in the config, that seems like a perfectly appropriate exception to use.

Reply:
Hm. I didn't consider this. I wanted to catch `PluginError`, more than anything else, and I still wanted something specific to both plugins & settings, which `ConfigurationError` isn't.

I thought that `ConfigurationError` couldn't carry the semantic required by the plugin. For instance, the config can tell you if the list is empty when it shouldn't, but it can't tell you that you have twice the same value, or that 2 values are mutually exclusive - this is more or less up to the plugin.

So, I don't know. I need to think about it.

Reply:
@dgw keep or drop? I still like it, but I decided to document the `url_lazy` decorator with the `PluginError` instead.

Reply:
@Exirel What if keep, but under a different name like `PluginConfigurationError` or `PluginSettingsError`?

Reply:
`PluginSettingsError` it is.